hazelcast/src/main/java/com/hazelcast/query/impl/OrderedIndexStore.java

```diff
@@ -138,21 +147,38 @@ public boolean canEvaluate(Class<? extends Predicate> predicateClass) {
     }
 
     @Override
-    public Iterator<QueryableEntry> getSqlRecordIterator(Comparison comparison, Comparable searchedValue) {
+    public Iterator<QueryableEntry> getSqlRecordIterator(Comparison comparison, Comparable searchedValue, boolean descending) {
         Iterator<Map<Data, QueryableEntry>> iterator;
 
+        ConcurrentNavigableMap navigableMap = descending ? recordMap.descendingMap() : recordMap;
         switch (comparison) {
             case LESS:
-                iterator = recordMap.headMap(searchedValue, false).values().iterator();
+                if (descending) {
```

Comment:
Consider the we have `LESS THAN EQUALS 2` operator and a collection of values `[0, 1, 2, 3, 4]`. 
For `ASC` we call `headMap(2, true)`, and get `[0, 1, 2]`. For `DESC` we call `tailMap(2, true)`, and get `[2, 3, 4]`. 

Looks like we return completely wrong results here unless I am missing something. Do we have tests for these cases?

Reply:
Nevermind, I missed the `navigableMap` initialization. Resolving.